### ISSUE TITLE
Rule proposal: Prefer `else`

### ISSUE THREAD

**Issue by fisker** [sindresorhus/eslint-plugin-unicorn#1128]
In following case, use `else` is more readable than `continue` and `return`, but only the `else` block is small.

## Fail

```js
for (const element of array) {
  if (foo) {
    bar();
    continue;
  }

  baz();
}
```

```js
function a() {
  if (foo) {
    bar();
    return;
  }

  baz();
}
```

## Pass

```js
for (const element of array) {
  if (foo) {
    bar();
  } else {
    baz();
  }
}
```

```js
function a() {
  if (foo) {
    bar();
  } else {
    baz();
  }
}
```

**Comment by sindresorhus** [comment]
Can you elaborate on why you think it's more readable? Because I personally find the `continue`/`return` version more readable, for the same reasons I prefer early-returns: less nesting.

**Comment by fregante** [comment]
This is the opposite of 

- https://github.com/sindresorhus/eslint-plugin-unicorn/issues/16
- https://github.com/Shopify/web-configs/blob/841aa44aee288ef809235c7e33c130708d66a6b9/packages/eslint-plugin/docs/rules/prefer-early-return.md

**Comment by fisker** [comment]
Thanks for your opinion.

**Comment by fregante** [comment]
The examples above are more readable with `else` IMHO, but that usually stops being the case once there's more than 1 line in the `else` block or if there's any additional block in `else`, e.g. 

```js
for (const element of array) {
  if (foo) {
    bar();
  } else {
    const he = baz();
	if (he.isCool) {
		he.call();
	}
  }
}
```

vs


```js
for (const element of array) {
  if (foo) {
    bar();
	continue;
  }

  const he = baz();
  if (he.isCool) {
    he.call();
  }
}
```

**Comment by fisker** [comment]
I agree this case  should do early exit.
